app/assets/javascripts/embed.js.erb

```diff
@@ -15,14 +15,7 @@ I18n.default_locale = <%= I18n.default_locale.to_json %>;
 I18n.fallbacks = true;
 
 window.onload = function () {
-  var query = (window.location.search || "?").slice(1),
-      args = {};
-
-  var pairs = query.split("&");
-  for (var i = 0; i < pairs.length; i++) {
-    var parts = pairs[i].split("=");
-    args[parts[0]] = decodeURIComponent(parts[1] || "");
-  }
+  var args = Object.fromEntries(new URLSearchParams(window.location.search).entries());
```

Comment:
Do you actually need `.entries()` here? I believe the `URLSearchParams` object itself should be iterable?